gha-webhook-listener.py

```diff
@@ -64,89 +70,88 @@ def create_symlink(source, linkname):
+        required_api_prefix = f'https://api.github.com/repos/{arg_github_org}/matrix.org/actions/artifacts'
 
     incoming_json = request.get_json()
     if not incoming_json:
         abort(400, "No JSON provided!")
         return
     logger.debug("Incoming JSON: %s", incoming_json)
 
-    event = incoming_json.get("event")
-    if event is None:
-        abort(400, "No 'event' specified")
+    action = incoming_json.get("action")
+    if action is None:
+        abort(400, "No 'action' specified")
         return
 
-    if event == 'ping':
-        logger.info("Got ping request - responding")
-        return jsonify({'response': 'pong!'})
+    if action == "completed":
+        logger.info("Workflow completed")
+    else:
+        abort(400, "Not a suitable event")
+        return
 
-    if event != 'build.finished':
+    workflow_status = incoming_json["workflow_run"]["conclusion"]
+    if workflow_status == "success":
+        logger.info("Workflow succeeded")
+    else:
         logger.info("Rejecting '%s' event", event)
         abort(400, "Unrecognised event")
         return
 
-    build_obj = incoming_json.get("build")
-    if build_obj is None:
-        abort(400, "No 'build' object")
+    build_id = incoming_json["workflow_run"]["id"]
+    if build_id is None:
+        abort(400, "No 'id' specified")
         return
 
-    build_url = build_obj.get('url')
-    if build_url is None:
-        abort(400, "build has no url")
+    workflow_id = incoming_json["workflow_run"]["workflow_id"]
+    if workflow_id is None:
+        abort(400, "No 'workflow_id' specified")
         return
 
-    if required_api_prefix is not None and not build_url.startswith(required_api_prefix):
-        logger.info("Denying poke for build url with incorrect prefix: %s", build_url)
-        abort(400, "Invalid build url")
+    artifacts_url = incoming_json["workflow_run"]["artifacts_url"]
+    if artifacts_url is None:
+        abort(400, "No 'artifacts_url' specified")
         return
 
-    build_num = build_obj.get('number')
-    if build_num is None:
-        abort(400, "build has no number")
-        return
-
-    pipeline_obj = incoming_json.get("pipeline")
-    if pipeline_obj is None:
-        abort(400, "No 'pipeline' object")
-        return
-
-    pipeline_name = pipeline_obj.get('slug')
-    if pipeline_name is None:
-        abort(400, "pipeline has no slug")
-        return
-
-    artifacts_url = build_url + "/artifacts"
-    logger.info("fetching %s", artifacts_url)
+    logger.info("Fetching %s", artifacts_url)
     artifacts_resp = requests.get(artifacts_url, headers=req_headers())
     artifacts_resp.raise_for_status()
-    artifacts_array = artifacts_resp.json()
+    artifacts_array = artifacts_resp.json()['artifacts']
 
     artifact_to_deploy = None
     for artifact in artifacts_array:
-        if re.match(arg_artifact_pattern, artifact['path']):
+        logging.info("Artifact name is %s", artifact['name'])
```

Comment:
This is going to the wrong logger, and I think we can downgrade it to `debug` and improve the wording:

```suggestion
                logger.debug("Considering artifact %s", artifact['name'])
```